### PR TITLE
Extend the release script to automatically push a new SyTest branch, rather than having that be a manual process.

### PR DESCRIPTION
**Pull request by reivilibre**
This is a typically error-prone part of our process, so it seems to make sense to try and automate it.

Should be a commit-by-commit review.

Unfortunately I haven't tested it myself yet, since I already did it :/.

Marking as a release blocker to try and give someone the idea that they should try it before creating the branch themselves.
It will only create the branch if it's also responsible for creating the branch on Synapse, **so you need to disregard the wiki instructions a bit and just run it straight on develop**.
(You probably need to make a copy of the script outside of the tree first, or you'll get told off for having a dirty tree)

### CODE REVIEW

**Comment by babolivier** [comment]
@reivilibre 

> Marking as a release blocker to try and give someone the idea that they should try it before creating the branch themselves

Marking it as a release blocker blocks it for v1.61.0 next week (or another RC if necessary). However, the `release-v1.61` branch already exists on SyTest, so I assume what we want to do is to make it a blocker for v1.62.0rc1 in two weeks? In this case I would suggest removing the label and re-adding it next week once 1.61.0 has shipped.

File: scripts-dev/release.py [hunk 1]
```diff
@@ -89,10 +92,12 @@ def prepare() -> None:
     """
 
     # Make sure we're in a git repo.
-    repo = get_repo_and_check_clean_checkout()
+    synapse_repo = get_repo_and_check_clean_checkout()
+    sytest_repo = get_repo_and_check_clean_checkout("../sytest", "sytest")
 
     click.secho("Updating git repo...")
```
Comment (anoadragon453) [inline scripts-dev/release.py:98]: As someone who needs to touch his YubiKey for each push/pull, it'd be good to be explicit about two `pull` operations here.

```suggestion
    click.secho("Updating Synapse and Sytest git repos...")
```

**Review by anoadragon453 — COMMENTED** [review]
There was an attempt to test this during the release of v1.62.0rc1, but it turned out that the `release-v1.62` branch had already been created on the Synapse. The branch must not exist for the script to try and create a Sytest branch with this PR.

We can try testing during the release of v1.63.0rc1 :)

File: scripts-dev/release.py (hunk 1 above)
Comment (reivilibre) [reply scripts-dev/release.py:98]: I had no idea that was a thing, but sure :-)

File: scripts-dev/release.py [hunk 2]
```diff
@@ -187,25 +192,36 @@ def prepare() -> None:
...
+            # See https://github.com/gitpython-developers/GitPython/pull/1419
+            repo.create_head(release_branch_name, commit=base_branch)  # type: ignore[arg-type]
+
+        # Special-case SyTest: we don't actually prepare any files so we may
+        # as well push it now (and only when we create a release branch;
+        # not on subsequent RCs or full releases).
+        if click.confirm("Push new SyTest branch?", default=True):
+            sytest_repo.git.push(
+                "-u", sytest_repo.remote().name, sytest_repo.active_branch.name
+            )
```
Comment (squahtx) [inline scripts-dev/release.py:220]: This didn't push the branch for me.

I started with sytest on a clean checkout of `squah/configure_codeowners`
and ended up on `HEAD -> develop, origin/develop, origin/HEAD, release-v1.63`, with `git status` saying I'm on `develop`.

Comment (reivilibre) [reply scripts-dev/release.py]: This was a typo, I believe this to be fixed since I tested it today.

**Comment by reivilibre** [comment]
Tested to work today with v1.64.0rc1 (`release-v1.64` branch on sytest)

**Review by squahtx — APPROVED** [review]
lgtm now that we've seen it work today!

**Pull request merged**